src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -7135,16 +7130,20 @@ internal static Expression InvokeMethod(MethodBase mi, DynamicMetaObject target,
 
                     if (expandParameters)
                     {
-                        argExprs[i] = Expression.NewArrayInit(
-                            paramElementType,
-                            args.Skip(i).Select(
-                                a => a.CastOrConvertMethodArgument(
+                        IEnumerable<Expression> elements = args
+                            .Skip(i)
+                            .Select(a =>
+                                a.CastOrConvertMethodArgument(
                                     paramElementType,
                                     paramName,
                                     mi.Name,
                                     allowCastingToByRefLikeType: false,
                                     temps,
-                                    initTemps)));
+                                    initTemps));
+
+                        argExprs[i] = Expression.NewArrayInit(paramElementType, elements);
+                        // User specified the element arguments, so we log them instead of the compiler-created array.
+                        argsToLog.AddRange(elements);
```

Comment:
Hmmm I'm worried about code executing twice, these may need to all be stored in temp locals. Let me build this PR locally and see how it disassembles. I'll update when I do

Reply:
If this will cause code executing twice, then the current way we are doing for logging has always been executing `args` twice ... :)

Reply:
Yep, `CastOrConvertMethodArgument` already creates temp locals, didn't expect that! I'll mark this resolved